SeaPublicWebsite/Services/RecommendationService.cs

```diff
@@ -1,130 +1,433 @@
+        {
+            int propertyType = GetConvertedPropertyType(userData.PropertyType);
+
+            (int builtForm, int? flatLevel) = GetConvertedBuiltFormAndFlatLevel(userData.PropertyType,
+                userData.HouseType,
+                userData.BungalowType, userData.FlatType);
+
+            string constructionDate = GetConvertedConstructionDate(userData.YearBuilt);
+
+            int wallType = GetConvertedWallType(userData.WallConstruction, userData.SolidWallsInsulated,
+                userData.CavityWallsInsulated);
+
+            int roofType = GetConvertedRoofType(userData.RoofConstruction, userData.RoofInsulated);
+
+            int glazingType = GetConvertedGlazingType(userData.GlazingType);
+
+            int heatingFuel = GetConvertedHeatingFuel(userData.HeatingType, userData.OtherHeatingType);
+
+            bool? hotWaterCylinder = GetConvertedHotWaterCylinder(userData.HasHotWaterCylinder);
+
+            int heatingPatternType = GetConvertedHeatingPatternType(userData.HeatingPattern);
+
+            BreRequest request = new()
             {
-                // Always shown
-                RecommendationKey.UpgradeHeatingControls
+                postcode = userData.Postcode,
+                property_type = propertyType.ToString(),
+                built_form = builtForm.ToString(),
+                flat_level = flatLevel.ToString(),
+                construction_date = constructionDate,
+                wall_type = wallType,
+                //no input for floor_type in BRE API
+                roof_type = roofType,
+                glazing_type = glazingType,
+                //no input for outdoor heater space in BRE API
+                heating_fuel = heatingFuel.ToString(),
+                hot_water_cylinder = hotWaterCylinder,
+                occupants = userData.NumberOfOccupants,
+                heating_pattern_type = heatingPatternType,
+                living_room_temperature = userData.Temperature,
+                //assumption:
+                num_storeys = userData.PropertyType == PropertyType.House ? 2 : 1,
+                //assumption:
+                num_bedrooms = userData.NumberOfOccupants ?? 1,
+                measures = true,
+                //measures_package consists of all measures implemented in the BRE API as of May 2021
+                measures_package = new[]
+                {
+                    "A", "A2", "B", "Q", "Q1", "W1", "W2", "D", "C", "F", "G", "I", "T", "L2", "N", "Y", "O", "O3", "X",
+                    "U"
+                }
             };
 
-            // Glazing is single or both
-            if (userData.GlazingType is GlazingType.Both or GlazingType.SingleGlazed)
+            return request;
+        }
+
+        private static int GetConvertedPropertyType(PropertyType? propertyType)
+        {
+            //ApartmentFlatOrMaisonette is assumed to be a Flat
+            if (propertyType != null) return (int) (PropertyTypeEnum) propertyType;
```

Comment:
This is relying on the two enums having the same int values for the same meaning, but there are no comments on the enums to warn anyone that changing them will break this mapping.
This code should really do an explicit switch statement so that the underlying values don't matter